src/components/furniture/Separators.vue

```diff
@@ -93,46 +94,58 @@ export default {
         height: this.shelf.height * 3 + 'px',
       };
     },
-    shelfIndex() {
-      return this.shelfs.findIndex((shelf) => shelf.id === this.shelf.id);
-    },
-    shelfSeparators() {
-      return this.shelfs[this.shelfIndex].insideSeparators;
-    },
     widthVariation() {
-      if (this.typeOfSeparators === 'centered') {
+      if (this.separators.typeOfSeparators === 'centered') {
         return 0;
-      } else if (this.typeOfSeparators === 'left') {
-        return -12 / this.amountOfSeparators;
+      } else if (this.separators.typeOfSeparators === 'left') {
+        return -12 / this.separators.amountOfSeparators;
       } else {
-        return 12 / this.amountOfSeparators;
+        return 12 / this.separators.amountOfSeparators;
       }
     },
     separatorWidth() {
-      if (this.amountOfSeparators === '1') {
+      if (this.separators.amountOfSeparators === '1') {
         return 50 + this.widthVariation;
-      } else if (this.amountOfSeparators === '2') {
+      } else if (this.separators.amountOfSeparators === '2') {
         return 33.3 + this.widthVariation;
-      } else if (this.amountOfSeparators === '3') {
+      } else if (this.separators.amountOfSeparators === '3') {
         return 25 + this.widthVariation;
       } else {
         return 100;
       }
     },
+    identifiedSeparators() {
+      const shelf = this.shelfs.find((shelf) => shelf.id === this.shelf.id);
+      return shelf.insideSeparators;
+    },
   },
-  watch: {
-    amountOfSeparators(amount) {
-      this.shelfSeparators.amountOfSeparators = amount;
-      this.shelfSeparators.widthInPercentaje = this.separatorWidth;
+  methods: {
+    updateSeparators() {
+      this.updateWidthBetweenSeparators();
+      this.updateSeparatorsInStore();
     },
-    typeOfSeparators(type) {
-      this.shelfSeparators.typeOfSeparators = type;
-      this.shelfSeparators.widthInPercentaje = this.separatorWidth;
+    updateWidthBetweenSeparators() {
+      this.separators.widthOfEachSpace = [];
+      this.pushSeparatorWidth();
+      this.pushRemainingWidth();
+    },
+    pushSeparatorWidth() {
+      for (let i = 0; i < this.separators.amountOfSeparators; i++) {
+        this.separators.widthOfEachSpace.push(this.separatorWidth);
+      }
+    },
+    pushRemainingWidth() {
+      this.separators.widthOfEachSpace.push(
+        100 -
+          this.separators.widthOfEachSpace.reduce(
+            (width, acc) => acc + width,
+            0
+          )
+      );
+    },
+    updateSeparatorsInStore() {
+      this.identifiedSeparators = this.separators;
```

Comment:
something is weird here, `this.identifiedSeparators` is a computed value (the product of the function `identifiedSeparators()`, so why are we assigning a value to this property? It should be computed but the function, not assigned.